core/src/main/java/org/apache/struts2/dispatcher/multipart/AbstractMultiPartRequest.java

```diff
@@ -169,4 +176,12 @@ protected String getCanonicalName(final String originalFileName) {
         return fileName;
     }
 
+    protected boolean isAccepted(String fileName) {
+        return patternsChecker.isAllowed(fileName).isAllowed();
+    }
+
+    protected String sanitizeNewlines(String before) {
```

Comment:
Nit: So I introduced this function in #760 but I realise now we can just use `org.apache.commons.lang3.StringUtils#normalizeSpace` instead

Reply:
Although given this is a protected method, deleting it would be a breaking change so only thing we could really do is replace the implementation with `return StringUtils.normalizeSpace(before);`

Reply:
We can always mark this function as deprecated and use `StringUtils.normalizeSpace()` directly

Reply:
Yep that works too